apps/meteor/client/components/AutoCompleteDepartment.tsx

```diff
@@ -1,18 +1,29 @@
-// Cannot convert this file to ts because PaginatedSelectFiltered is not typed yet
-// Next release we'll add required types and convert this file, since a new
-// fuselage release is OoS of this regression
 import { PaginatedSelectFiltered } from '@rocket.chat/fuselage';
 import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
 import { useTranslation } from '@rocket.chat/ui-contexts';
-import React, { memo, useMemo, useState } from 'react';
+import React, { memo, ReactElement, useMemo, useState } from 'react';
 
 import { useRecordList } from '../hooks/lists/useRecordList';
 import { AsyncStatePhase } from '../hooks/useAsyncState';
 import { useDepartmentsList } from './Omnichannel/hooks/useDepartmentsList';
 
-const AutoCompleteDepartment = (props) => {
-	const { value, excludeDepartmentId, onlyMyDepartments = false, onChange = () => {}, haveAll = false, haveNone = false } = props;
+type AutoCompleteDepartmentProps = {
+	value: { value: string; label: string } | string;
+	onChange: (value: string) => void;
+	excludeDepartmentId?: string;
+	onlyMyDepartments?: boolean;
+	haveAll?: boolean;
+	haveNone?: boolean;
+};
 
+export const AutoCompleteDepartment = ({
+	value,
+	excludeDepartmentId,
+	onlyMyDepartments,
+	onChange,
+	haveAll,
+	haveNone,
+}: AutoCompleteDepartmentProps): ReactElement | null /* null added to satisfy register type condition */ => {
```

Comment:
I think it is, because there is no null return as conditional at the component... but, as register.ts has a dynamic import there... it makes sense to have this null for default exports

Reply:
well, removed...